examples/nlp/Tweet-classification-using-TFDF.py

```diff
@@ -0,0 +1,215 @@
+train_df = pd.read_csv(
+    "https://raw.githubusercontent.com/IMvision12/Tweets-Classification-NLP/main/train.csv"
+)
+print(train_df.head())
+
+# Printing shape of the training dataset
+print(train_df.shape)
+
+# Shuffling training data
+train_df_shuffled = train_df.sample(frac=1, random_state=42)
+# Dropping id, keyword and location columns as these columns consists of mostly nan values
+# we will be using only text and target columns
+train_df_shuffled.drop(["id", "keyword", "location"], axis=1, inplace=True)
+train_df_shuffled.reset_index(inplace=True, drop=True)
+print(train_df_shuffled.head())
+
+print(train_df_shuffled.info())
+
+# Printing total number of Disaster and non-Disaster tweets
+print(train_df_shuffled.target.value_counts())
+
+# Viewing 5 records from training data
+for ind, counter in enumerate(train_df_shuffled.index):
+    print(f"Target : {train_df_shuffled['target'][ind]}")
+    print(f"Text : {train_df_shuffled['text'][ind]}")
+    if counter == 5:
+        break
+
+# Splitting dataset into train and test
+test_df = train_df_shuffled.sample(frac=0.2, random_state=42)
+train_df = train_df_shuffled.drop(test_df.index)
+
+print(f"Using {len(train_df)} samples for training and {len(test_df)} for validation")
+
+"""
+## Convert data to tf.Dataset
+"""
+
+
+def create_dataset(dataframe, training=False):
+    df = dataframe.copy()
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (df["text"].to_numpy(), df["target"].to_numpy())
+    )
+
+    if training:
+        dataset = dataset.repeat()
+        dataset = dataset.shuffle(1000)
+
+    dataset = dataset.batch(32)
+    dataset = dataset.prefetch(tf.data.AUTOTUNE)
+    return dataset
+
+
+train_ds = create_dataset(train_df, training=True)
+test_ds = create_dataset(test_df)
+
+"""
+## Downloading Pre-trained Embeddings
+
+The Universal Sentence Encoder encodes text into high-dimensional vectors that can be
+used for text classification, semantic similarity, clustering and other natural language
+tasks.It is trained on a variety of data sources and a variety of tasks. The input is
+variable length English text and the output is a 512 dimensional vector.
+
+To understand better about pre-trained embeddings : [Universal Sentence
+Encoder](https://tfhub.dev/google/universal-sentence-encoder/4)
+
+
+"""
+
+sentence_encoder_layer = hub.KerasLayer(
+    "https://tfhub.dev/google/universal-sentence-encoder/4",
+    input_shape=[],
+    dtype=tf.string,
+    trainable=False,
+    name="USE",
+)
+
+"""
+## Build a model
+
+First, we develop a preprocessor model that takes the inputs and passes it through
+sentence encoder layer to generate embedding, which produces a 512 dimensional vector for
+each sentence.
+
+The Preprocessor is then passed to the GradientBoostedTreesModel which will be used for
+predicting disaster tweets.
+
+
+"""
+
+inputs = layers.Input(shape=(), dtype=tf.string)
+# Passing sentences to sentence_encoder_layer
+x = sentence_encoder_layer(inputs)
```

Comment:
Nit: You can also concatenate the raw text feature to the embedding. This way, the TF-DF model consumes both the pre-trained embedding and the raw text.

Reply:
@achoum That's a great idea but how do i do that, if I cannot concatenate raw text with embeddings there will be a shape error